### PR TITLE
documentation: Point README to developer guide

### PR DESCRIPTION
**Pull request by speller26**
Instead of listing available devices, the README now links to the Amazon
Braket developer guide. The prevents the need to change the README with
each new device.

Also changed the bucket used in examples from amazon-braket-output-<ID>
to amazon-braket-<ID>, the correct default bucket name.

*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,11 +91,11 @@ print(batch.results()[0].measurement_counts)  # The result of the first task in
...
+Two types of circuit simulators are available: fully managed simulators accessed through the Amazon Braket service, and a local simulator bundled with the Amazon Braket SDK.
 
-- `arn:aws:braket:::device/quantum-simulator/amazon/sv1` - SV1 is a fully managed, high-performance, state vector simulator. It can simulate circuits of up to 34 qubits and has a maximum runtime of 12h. You should expect a 34-qubit, dense, and square (circuit depth = 34) circuit to take approximately 1-2 hours to complete, depending on the type of gates used and other factors.
+- The fully managed simulators offer high performance circuit simulations, and can handle larger circuits than can be run on user hardware. For example, the SV1 state vector simulator used in the above examples will take approximately 1-2 hours to complete to run a 34-qubit, dense, and square (circuit depth = 34) circuit, depending on the type of gates used and other factors. For a list of available simulators and their features, consult the Amazon Braket [developer guide](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html).
```
Comment (wrasmuss) [inline README.md:96]:
```suggestion
- The fully managed simulators offer high performance circuit simulations, and can handle larger circuits than can be run on user hardware. For example, the SV1 state vector simulator used in the above examples will take approximately 1-2 hours to complete to run a 34-qubit, dense, and square (circuit depth = 34) circuit, depending on the type of gates used and other factors. For a list of available simulators and their features, consult the Amazon Braket [developer guide full device list](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html).
```

**Review by wrasmuss — COMMENTED** [review]
See single suggestion

File: README.md [hunk 2]
```diff
@@ -128,12 +128,7 @@ task = device.run(bell, s3_folder, poll_timeout_seconds=86400)  # 1 day
 print(task.result().measurement_counts)
 ```
 
-Specify which quantum computer hardware to use by changing the value of the `device_arn` to the value for quantum computer to use:
-- **IonQ** "arn:aws:braket:::device/qpu/ionq/ionQdevice"
-- **Rigetti** "arn:aws:braket:::device/qpu/rigetti/Aspen-8"
-- **D-Wave** (See the next section in this document for more information about using D-Wave.)
-    - **D-Wave 2000Q_6** "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6"
-    - **D-Wave Advantage_system1** "arn:aws:braket:::device/qpu/d-wave/Advantage_system1"
+Specify which quantum hardware device to use by specifying its ARN as the value of the `device_arn` argument. A list of all available quantum devices and their features can be found in the Amazon Braket [developer guide](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html).
```
Comment (lundql) [inline README.md:131]: To select a quantum hardware device, specify its ARN as the value of the `device_arn` argument. A list of available quantum devices and their features can be found in the [Amazon Braket Developer Guide](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html).

(Note to writer: The actual title of the document is as shown. Other edits for clarity and grammar.)

**Review by lundql — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -91,11 +91,11 @@ print(batch.results()[0].measurement_counts)  # The result of the first task in
 ```
 
 ### Available Simulators
-Amazon Braket provides access to two simulators: a fully managed statevector simulator, SV1, and a local simulator that is part of the Amazon Braket SDK.
+Two types of circuit simulators are available: fully managed simulators accessed through the Amazon Braket service, and a local simulator bundled with the Amazon Braket SDK.
```
[the remark this answers is not in the document]
Comment (speller26) [reply README.md:94]: The following long descriptions are also bulleted; how should they be arranged relative to these bullets?

[the remark this answers is not in the document]
Comment (lundql) [reply README.md:94]: I edited the previous comment per your question

**Review by lundql — APPROVED** [review]

**Pull request merged**